lib/HappyPlugin.js

```diff
@@ -340,11 +340,9 @@ function compileAndUpdateCache(threadPool, loader, loaderContext, done) {
   var cache = this.cache;
   var filePath = loaderContext.resourcePath;
 
-  cache.invalidateEntryFor(filePath);
-
   threadPool.compile(loaderContext.remoteLoaderId, loader, {
     loaders: this.state.loaders,
-    compiledPath: path.resolve(this.config.tempDir, HappyUtils.generateCompiledPath(filePath)),
+    compiledPath: path.resolve(this.config.tempDir, cache.getCompiledSourceCodePath(filePath) || HappyUtils.generateCompiledPath(filePath)),
```

Comment:
Argh... this is because of a [recent change to `#generateCompiledPath`](https://github.com/amireh/happypack/commit/6b60e14e604e925893cf8ef5e702ba2805398b5d#diff-ac8b1b3b97f21cbb24e176aab19ad0cd) that made it no longer re-entrant. The original solution was such that for any given file path, an identical "compiled" file path was produced. That's why this code was never needed.

This is a bug that should be fixed and I don't think this is the right place for it. I would like to unroll that change and either make an option to make the cache path randomizable, or figure out some other way.